packages/gui/src/init-prefs.ts

```diff
@@ -0,0 +1,69 @@
+import type { IpcRenderer } from 'electron';
+
+type WindowExt = typeof window & {
+  preferences: Record<string, any>;
+  ipcRenderer: IpcRenderer;
+};
+
+export async function initPrefs(onInitCallback: Function){
+  const w = window as WindowExt;
+  await migrateGUIPrefsFromLocalStorage();
+  w.preferences = await w.ipcRenderer.invoke('readPrefs');
+  onInitCallback();
+}
+
+async function migrateGUIPrefsFromLocalStorage(){
+  const w = window as WindowExt;
+  const prefs: Record<string, any> = {};
+  const targets: string[] = [
+    'cacheFolder',
+    'cacheLimitSize',
+    'darkMode',
+    'enableAutoLogin',
+    'enableDataLayerService',
+    'enableFilePropagationServer',
+    'finaldir',
+    'fingerprintSettings',
+    'hiddenWalletsItems',
+    'hideObjectionableContent',
+    'isHidden',
+    'limit-cache-size',
+    'locale',
+    'mode',
+    'sensitive-content',
+    'skipMigration',
+    'suppressShareOnCreate',
+    'suppressUnsafeLinkWarning',
+    'tmp2dir',
+    'tmpdir',
+  ];
+  // Items which were not stringified on localStorage
+  const noParse: string[] = [
+    'finaldir',
+    'tmp2dir',
+    'tmpdir',
+  ];
+
+  for(const key of targets){
+    const item = window.localStorage.getItem(key);
+    if(item === undefined || item === null){
+      continue;
+    }
+
+    try{
+      prefs[key] = noParse.includes(key) ? item : JSON.parse(item);
+      window.localStorage.removeItem(key);
```

Comment:
Can we remove the items from localStorage _after_ `migratePrefs` is called successfully? If there is a problem executing migratePrefs, the user will end up with their prefs being cleared.

Reply:
Thanks for checking. You're right. I updated the code as you suggested.